### ISSUE TITLE
Mark vgettext() APIs as deprecated or implementation details

### ISSUE THREAD

**Issue by shikadiqueen** [wesnoth/wesnoth#2716]
As I extensively explained in PR #2711, the `vgettext()` and `vngettext()` APIs are unsuitable for general use because one of their overloads silently force a textdomain that may not be the one expected by the caller, and the other is incorrectly parsed by `xgettext` (and is also too verbose), resulting in localization issues at runtime. The `VGETTEXT()` and `VNGETTEXT()` macros should be used instead, as they take care of passing the correct `GETTEXT_DOMAIN` value.

@CelticMinstrel suggests marking the functions as deprecated when directly called rather than invoked through the aforementioned wrapper macros. My suggestion is simply to rename them both `xxxx_internal_()` or so and add documentation in the relevant header to make it clearer to coders that they should not be used directly without extreme care. Either way works for me, since the only goal here is to keep devs from silently introducing issues that are impossible to spot when using the default English locale.

I'd also dare suggest that the overloads that internally enforce a textdomain (currently wesnoth-lib) instead of letting the caller specify one should be culled as they do not seem to be needed by anything after #2711 and its upcoming port to master.

(Also, in case it's not obvious, to avoid noise and breakage potential, this change should obviously only be applied to master, not 1.14.)

**Comment by CelticMinstrel** [comment]
My only complaint is they incorrect use of Doxygen comments.

**Comment by Vultraz** [comment]
Is one supposed to use `/*` for general-purpose comments?

**Comment by shikadiqueen** [comment]
`/**` is not part of the C++ syntax, `/*` and `//` are, so yes. Comments starting with `/**`, `/**<`, `///` and `///<` are for Doxygen's consumption.

**Comment by CelticMinstrel** [comment]
The main issue is that one comment cannot serve two functions.

You can however use `@copydoc otherFunction()` to avoid duplicate documentation of the functions really are the same.
